conformance/lang/expressions/logical-expr/logical_and_expr.balt

```diff
@@ -0,0 +1,849 @@
+Labels: logical-expr, string-template-expr, string
+
+function errorFunction() {
+    string isTrue = "true";
+    boolean _ = true && string `test ${isTrue}`; // @error static type of each expression must be a subtype of boolean
+    boolean _ = string `test ${isTrue}` && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = string `test ${isTrue}` && string `test ${isTrue}`; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using xml template expression.
+Labels: logical-expr, xml-template-expr
+
+function errorFunction() {
+    boolean _ = true && xml `false`; // @error static type of each expression must be a subtype of boolean
+    boolean _ = xml `true` && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = xml `true` && xml `false`; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using raw template expression.
+Labels: logical-expr, raw-template-expr, string
+
+function errorFunction() {
+    string a = "true";
+    string b = "false";
+    boolean _ = true && `rawTemplate ${a}`; // @error static type of each expression must be a subtype of boolean
+    boolean _ = `rawTemplate ${b}` && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = `rawTemplate ${b}` && `rawTemplate ${a}`; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using structural constructor expression.
+Labels: logical-expr, list-constructor-expr
+
+function errorFunction() {
+    boolean _ = true && [1, 2, 3]; // @error static type of each expression must be a subtype of boolean
+    boolean _ = [1, 2, 3] && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = [1, 2, 3] && [1, 7, 3]; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using new expression.
+Labels: logical-expr, new-expr, module-class-defn
+
+class Person {
+    int a = 0;
+
+    function init() {
+        self.a = 4;
+    }
+}
+
+function errorFunction() {
+    boolean _ = true && new Person(); // @error static type of each expression must be a subtype of boolean
+    boolean _ = new Person() && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = new Person() && new Person(); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using variable reference expression.
```

Comment:
We already have variable reference tests above. Let's either remove this test or specifically say "variable reference of simple basic type".

Reply:
[94d5da7](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/94d5da7d6a3c741d8d2a8cb2d22ac6910b20f776)